ansible/include_vars.yml

```diff
@@ -1,58 +1,115 @@
+          include_vars:
+            file: "{{ item.stat.path }}"
+          when:
+            - item is not skipped
+            - item.stat.exists
+          loop: "{{ rstat_varfiles.results + rstat2_varfiles.results }}"
+          loop_control:
+            label: >-
+              {{ (
+                item.stat.path
+                | default('skipped')
+              ) if 'stat' in item else item }}
+      when: hostvars[inventory_hostname].ostype | d('linux') != 'windows'
 
-    - name: Include variables files
-      include_vars:
-        file: "{{ item.stat.path }}"
+      # TODO (jbirchler) This current block enables support for Windows
+      # systems. Is it still necessary?
+    - name: Stat default variables files (both yaml/yml extensions) [Windows]
+      block:
+        - name: stat environment variables
+          vars:
+            find_me:
+              # Global default vars related to the cloud provider
+              - cloud_providers/{{ cloud_provider }}_default_vars
+              # Legacy env_vars.yml (replaced by default_vars.yml)
+              - configs/{{ env_type }}/env_vars
+              # Default vars of the config
+              - configs/{{ env_type }}/default_vars
+              # Default vars of the config, specific to a cloud provider
+              - configs/{{ env_type }}/default_vars.{{ cloud_provider }}
+              - configs/{{ env_type }}/default_vars_{{ cloud_provider }}
+              # Legacy secret vars file.
+              - configs/{{ env_type }}/env_secret_vars
+            extensions:
+              - yaml
+              - yml
+          stat:
+            path: "{{ item[0] ~ '.' ~ item[1] }}"
+          loop: "{{ find_me | product(extensions) | list }}"
+          register: rstat_varfiles
+
+        - name: Stat variables files
+          vars:
+            find_me:
+              # secret file path passed as extra-var
+              - "{{ secret_file | d('/secret/file/not/passed') }}"
+          stat:
+            path: "{{ item }}"
+          loop: "{{ find_me }}"
+          register: rstat2_varfiles
+
+        - name: Include variables files
+          include_vars:
+            file: "{{ item.stat.path }}"
+          when:
+            - item is not skipped
+            - item.stat.exists
+          loop: "{{ rstat_varfiles.results + rstat2_varfiles.results }}"
+          loop_control:
+            label: >-
+              {{ (
+                item.stat.path
+                | default('skipped')
+              ) if 'stat' in item else item }}
       when:
-        - item is not skipped
-        - item.stat.exists
-      loop: "{{ rstat_varfiles.results + rstat2_varfiles.results }}"
-      loop_control:
-        label: >-
-          {{ (
-            item.stat.path
-            | default('skipped')
-          ) if 'stat' in item else item }}
+        - groups['windows'] is defined
+        - inventory_hostname in groups['windows']
+      delegate_to: localhost
```

Comment:
Ah! So `delegate_to: localhost` might be workaround to a problem locating the correct python/venv?

> "The problem with connection: local for the runner pattern is that it assumes that it’s an entirely new connection and will use the system python rather than what ever python you prefer to use. Situations where this is a problem include when using virtualenvs to install python libraries or on OS X where most rely on python from brew. In this case, you might run pip install library, run Ansible and find that that library can’t be found because it’s looking in the wrong place."
[source](http://willthames.github.io/2018/07/01/connection-local-vs-delegate_to-localhost.html)

What if instead of duplicating everything and running windows hosts with `delegate_to: localhost`, we just ran everything with `delegate_to: localhost`? (Obviously will need some testing.)

Reply:
Jim says problem was with windows and variables not being set.